lib/resolvers/review.ts

```diff
@@ -0,0 +1,68 @@
+import { createReviewLoader } from "../dataloaders/reviewLoader";
+import { Review } from "../models/review";
+
+const reviewLoader = createReviewLoader();
+
+export async function reviewQueryResolver(
+  _parent: unknown,
+  {
+    bookId,
+    id,
+    limit = 10,
+    offset = 0,
+  }: {
+    bookId?: string;
+    id?: string;
+    limit?: number;
+    offset?: number;
+  }
+) {
+  if (id) {
+    const review = await reviewLoader.load(id);
+    return {
+      reviews: review ? [review] : [],
+      count: review ? 1 : 0,
+    };
+  }
+  const query: any = {};
+  if (bookId) query.bookId = bookId;
+
+  const count = await Review.countDocuments(query);
+  const reviews = await Review.find(query)
+    .sort({ createdAt: -1 })
+    .skip(offset)
+    .limit(limit);
+
+  return {
+    reviews,
+    count,
+  };
+}
```

Comment:
Would be useful to move the Module level functions, which are working with storage, to different files, that would allow reusability in different resolvers + Mutation. 

Like we create a function called `getBooksPaginated` in a file called `BookModule` that can be used in differnt places.